Python package set major updates

FRidh opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

tbenst committed:
pythonPackages.tifffile: add missing dependency

(cherry picked from commit ba31a6835df40677095ffa952d3a2e07c95555a0)

FRidh committed:
python: pydantic: 1.0 -> 1.2

FRidh committed:
python: pycares: 3.0.0 -> 3.1.0

FRidh committed:
python: pybullet: 2.5.8 -> 2.6.0

FRidh committed:
python: pybids: 0.9.5 -> 0.10.0

FRidh committed:
python: pyro-ppl: 0.5.1 -> 1.1.0

FRidh committed:
python: pymongo: 3.9.0 -> 3.10.0

FRidh committed:
python: pykeepass: 3.0.3 -> 3.1.2

FRidh committed:
python: pyOpenSSL: 19.0.0 -> 19.1.0

FRidh committed:
python: pymssql: 2.1.4 -> 3.0.3

FRidh committed:
python: pyfakefs: 3.6.1 -> 3.7

FRidh committed:
python: pyobjc: 6.0.1 -> 6.1

FRidh committed:
python: pyelftools: 0.25 -> 0.26

FRidh committed:
python: pymatgen: 2019.10.16 -> 2019.12.3

FRidh committed:
python: pymc3: 3.7 -> 3.8

FRidh committed:
python: pyperf: 1.6.1 -> 1.7.0

FRidh committed:
python: pytest-services: 1.3.1 -> 2.0.1

FRidh committed:
python: pytest-testmon: 0.9.19 -> 1.0.1

FRidh committed:
python: pytest-mock: 1.11.2 -> 1.13.0

FRidh committed:
python: python-gitlab: 1.13.0 -> 1.15.0

FRidh committed:
python: pytest-runner: 5.1 -> 5.2

FRidh committed:
python: pyscreenshot: 0.5.1 -> 0.6

FRidh committed:
python: pytest-doctestplus: 0.4.0 -> 0.5.0

FRidh committed:
python: pytest-qt: 3.2.2 -> 3.3.0

FRidh committed:
python: pytest-mpl: 0.10 -> 0.11

FRidh committed:
python: pytest-django: 3.6.0 -> 3.7.0

FRidh committed:
python: pytest: 5.2.4 -> 5.3.2

FRidh committed:
python: python-socketio: 4.3.1 -> 4.4.0

FRidh committed:
python: pyxl3: 1.2 -> 1.3

FRidh committed:
python: python-redis-lock: 3.3.1 -> 3.4.0

FRidh committed:
python: pyxattr: 0.6.1 -> 0.7.1

FRidh committed:
python: pywinrm: 0.3.0 -> 0.4.1

FRidh committed:
python: python2-pythondialog: 3.4.0 -> 3.5.1

FRidh committed:
python: pywbem: 0.14.6 -> 0.15.0

FRidh committed:
python: pyunifi: 2.18 -> 2.19.0

FRidh committed:
python: PyYAML: 5.1.2 -> 5.2

FRidh committed:
python: python-rapidjson: 0.8.0 -> 0.9.1

FRidh committed:
python: pythondialog: 3.5.0 -> 3.5.1

FRidh committed:
python: s3fs: 0.3.5 -> 0.4.0

FRidh committed:
python: responses: 0.10.7 -> 0.10.8

FRidh committed:
python: shodan: 1.20.0 -> 1.21.0

FRidh committed:
python: Scrapy: 1.7.4 -> 1.8.0

FRidh committed:
python: rpy2: 3.2.2 -> 3.2.4

FRidh committed:
python: regex: 2019.08.19 -> 2019.12.19

FRidh committed:
python: six: 1.12.0 -> 1.13.0

FRidh committed:
python: qiskit: 0.13.0 -> 0.14.0

FRidh committed:
python: rainbowstream: 1.4.0 -> 1.5.2

FRidh committed:
python: rlp: 1.1.0 -> 1.2.0

FRidh committed:
python: qtconsole: 4.5.5 -> 4.6.0

FRidh committed:
python: slicedimage: 4.0.5 -> 4.1.1

FRidh committed:
python: tblib: 1.5.0 -> 1.6.0

FRidh committed:
python: sphinxcontrib-blockdiag: 1.5.5 -> 2.0.0

FRidh committed:
python: tenacity: 5.1.5 -> 6.0.0

FRidh committed:
python: starlette: 0.12.13 -> 0.13.0

FRidh committed:
python: spyder: 3.3.6 -> 4.0.0

FRidh committed:
python: splinter: 0.11.0 -> 0.13.0

FRidh committed:
python: spyder-kernels: 0.5.2 -> 1.8.1

FRidh committed:
python: SQLObject: 3.7.3 -> 3.8.0

FRidh committed:
python: slither-analyzer: 0.3.0 -> 0.6.8

FRidh committed:
python: starfish: 0.1.9 -> 0.1.10

FRidh committed:
python: stripe: 2.37.2 -> 2.41.0

FRidh committed:
python: svg.path: 3.0 -> 4.0.2

FRidh committed:
python: thrift: 0.11.0 -> 0.13.0

FRidh committed:
python: trio: 0.12.1 -> 0.13.0

FRidh committed:
python: typeguard: 2.5.1 -> 2.7.0

FRidh committed:
python: trustme: 0.5.3 -> 0.6.0

FRidh committed:
python: uncompyle6: 3.5.1 -> 3.6.0

FRidh committed:
python: twine: 2.0.0 -> 3.1.1

FRidh committed:
python: tld: 0.9.8 -> 0.11.9

FRidh committed:
python: Twisted: 19.7.0 -> 19.10.0

FRidh committed:
python: uproot: 3.10.12 -> 3.11.0

FRidh committed:
python: uamqp: 1.2.4 -> 1.2.5

FRidh committed:
python: uproot-methods: 0.7.1 -> 0.7.2

FRidh committed:
python: transaction: 2.4.0 -> 3.0.0

FRidh committed:
python: venusian: 1.2.0 -> 3.0.0

FRidh committed:
python: yapf: 0.28.0 -> 0.29.0

FRidh committed:
python: vcrpy: 2.1.1 -> 3.0.0

FRidh committed:
python: whitenoise: 4.1.4 -> 5.0.1

FRidh committed:
python: urwid: 2.0.1 -> 2.1.0

FRidh committed:
python: PyVirtualDisplay: 0.2.4 -> 0.2.5

FRidh committed:
python: vowpalwabbit: 8.7.0.post1 -> 8.8.0

FRidh committed:
python: virtualenv: 16.7.8 -> 16.7.9

FRidh committed:
python: xml2rfc: 2.35.0 -> 2.37.2

FRidh committed:
python: vega_datasets: 0.7.0 -> 0.8.0

FRidh committed:
python: uritools: 2.2.0 -> 3.0.0

FRidh committed:
python: zope.interface: 4.6.0 -> 4.7.1

FRidh committed:
python: yarl: 1.3.0 -> 1.4.2

FRidh committed:
python: yappi: 1.0 -> 1.2.3

FRidh committed:
python: zm-py: 0.3.3 -> 0.4.0

FRidh committed:
python: zeroconf: 0.23.0 -> 0.24.2

FRidh committed:
python: zope.component: 4.5 -> 4.6

FRidh committed:
python: yubico-client: 1.11.0 -> 1.12.0

FRidh committed:
python: zope.contenttype: 4.4 -> 4.5.0

FRidh committed:
python: zope.i18nmessageid: 4.3.1 -> 5.0.0

FRidh commented:
@jonringer feel free to force-push.

FRidh commented:
Should also consider https://github.com/NixOS/nixpkgs/pull/65023 again

marsam commented:
can you revert the gym update https://github.com/NixOS/nixpkgs/pull/76059/commits/834a4f76d2aa951f93faa41e9980352f3327be9e?, because it requires python-opencv which is not in nixpkgs yet

FRidh commented:
Done

FRidh commented:
Could you refer me to that `python-opencv`. It's a mess with all those different `opencv` packages.

marsam commented:
Thanks.
It's https://pypi.org/project/opencv-python/ https://github.com/skvark/opencv-python 
I recall @jonringer mentioning that packaging it was not a trivial task

jonringer commented:
IIRC, we can also use the opencv package to make the bindings, the https://github.com/skvark/opencv-python repo is just there to publish the wheels.

But tying to build from opencv-python is painful as it makes a lot of assumptions about network access, git, and cmake integration.

FRidh commented:
Yes, we just need to build a dist-info folder that has the `opencv-python` name.

jonringer commented:
the only packages i can think of that use opencv-python are imgaug, baselines, and gym

jonringer commented:
there's a PR to fix imgaug here: https://github.com/NixOS/nixpkgs/pull/67494
gym doesn't need it unless it's bumped
and baselines currently removes it from the install_requires

Although eventually, we should add it. I just don't have the will right now.

jonringer commented:
did you want help stabilizing this? because I know a lot of leaf packages probably got broken

FRidh commented:
Yes, I think we need to do such an upgrade again. I saw some of the PR's, and those are bound to cause regressions as well. With this PR/branch we can test them.

I will actually be away for a week so probably need to rerun this after.

Commit:
pythonPackages.urwid: enable glibclocales for tests

Commit:
pythonPackages.urwidtrees: enable glibclocales for tests

Commit:
importlib-metadata: 0.23 -> 1.3.0

Commit:
pythonPackages.html5lib: use pytest_4

Commit:
pythonPackages.ecdsa: add six dependency

Commit:
pythonPackages.aiohttp: disable sandbox-incompatible test

Commit:
pythonPackages.paramiko: fix build

jonringer committed:
pythonPackages.backports_functools_lru_cache: fix tests

jonringer committed:
pythonPackages.inflect: add importlib-metadata dependency

jonringer committed:
pythonPackages.jaraco_classes: add more-itertools dependency

jonringer committed:
pythonPackages.natsort: ignore impure tests

jonringer commented:
I feel like now would be a good time to bump pytest and hypothesis

jonringer committed:
python3Packages.azure-core: init at 1.1.1

jonringer committed:
python3Packages.azure-keyvault-keys: init at 4.0.0

jonringer committed:
python3Packages.azure-keyvault-secrets: init at 4.0.0

jonringer committed:
python3Packages.azure-keyvault: fix build

jonringer committed:
python3Packages.flask-caching: add missing pytestcache dependency

jonringer committed:
azure-cli: 2.0.77 -> 2.0.78

jonringer committed:
python3Packages.pyfakefs: disable tests for python3.7

the glob mocks seem to be unable to handle compile_pattern

jonringer committed:
python3Packages.keyringer: add importlib-metadata dependency

jonringer committed:
python3Packages.pgcli: remove prompt_tookit version bounds

jonringer committed:
python3Packages.bleach: disable network tests

jonringer committed:
python3Packages.venusian: add pytestcov test dependency

jonringer committed:
pythonPackages.cachetools: disable py2, abandoned upstream

jonringer committed:
pythonPackages.squaremap: mark broken

wxPython is unable to be detected by setuptools

jonringer committed:
python3Packages.hypothesis: 4.41.0 -> 4.56.1

jonringer committed:
python3Packages.cymem: 2.0.2 -> 2.0.3

jonringer committed:
python3Packages.ipykernel: 5.1.2 -> 5.1.3

jonringer committed:
python3Packages.logilab_common: 1.4.4 -> 1.5.2

jonringer committed:
python3Packages.pysocks: 1.7.0 -> 1.7.1

jonringer committed:
python3Packages.simplejson: 3.16.1 -> 3.17.0

jonringer committed:
pytphon3Packages.sphinx: 2.2.0 -> 2.2.2

FRidh commented:
> I feel like now would be a good time to bump pytest and hypothesis

Definitely!

Thank you for looking at these packages. Next Sunday I'll be back and can have a look as well.

jonringer committed:
python3Packages.cfn-lint: fix build

jonringer committed:
pythonPackages.logilab_common: disable for python2

jonringer committed:
pythonPackages.munch: add missing pbr dependency

jonringer commented:
had to revert sip bump, as it broke a lot of pyqt builds

jonringer commented:
going to revert the boto/moto bumps apparently aws is a very fragile api

jonringer commented:
also reverted the major coverage bump, as that broke quite a few packages

jonringer commented:
did you want to bump python3.{6,7,8} on this branch? @FRidh

jonringer commented:
https://www.python.org/downloads/ they all got a bump on the same day

jonringer commented:
python3.9a2 was also released https://www.python.org/downloads/release/python-390a2/

jonringer committed:
python3Packages.pytest-services: add missing zc_lockfile dependency

jonringer committed:
python3Packages.pytest-testmon: only run tests which don't import unittest_mixins

jonringer committed:
pythonPackages.libcloud: fix build

jonringer committed:
python3Packages.argon2_cffi: fix src pname

jonringer committed:
python3Packages.portend: fix test dependencies

jonringer committed:
pythonPackages.keyrings-alt: disable tests

They heavily rely on the keyring package test suite

jonringer committed:
pythonPackages.pycurl: fix tests

jonringer committed:
pythonPackages.bonfire: mark broken

arrow package now only allows for time attributes
to be in the singular form (minute vs minutes).
bonfire also does not seem to be actively maintained

jonringer committed:
pythonPackages.nibabel: disable python2 version

jonringer committed:
python3Packages.nbsmoke: add missing holoviews dependency

jonringer committed:
pythonPackages.solo-python: mark disabled

no longer compatible with latest fido2

jonringer committed:
pythonPackages.readme_renderer: disable flakey test

jonringer committed:
pythonPackages.celery: fix tests

jonringer committed:
pythonPackages.whitenoise: python2 no longer supported

jonringer commented:
had to revert flit major bump. they broke it into a flit and flit_core package, but they both require flit to build a wheel

jonringer committed:
pythonPackages.marshmallow: python2 no longer supported

jonringer committed:
pythonPackages.gunicorn: python2 no longer supported

jonringer committed:
pythonPackages.datadog: add missing freeze test dep

jonringer committed:
python3Packages.twine: fix build deps

jonringer committed:
python3Packages.pytest-mpl: add missing pillow dep

jonringer committed:
pythonPackages.audio-metadata: adjust more-itertools version range

jonringer commented:
reverted pytest-astropy, added a new pytest-astropy-headers dependency, which doesn't exist yet

jonringer committed:
pythonPackages.xdis: mark broken

Very specific to patch version of python interpreter,
unfortunately doesn't support newest versions

jonringer committed:
pythonPackages.mockito: disable tests

no longer packaged on pypi

jonringer committed:
python3Packages.uritools: python2 no longer supported

jonringer committed:
pythonPackages.djangorestframework: python2 no longer supported

jonringer committed:
pythonPackages.jupyter_console: fix prompt_toolkit vers bounds

jonringer committed:
awscli: fix build

jonringer committed:
aws_shell: fix dep issues

jonringer committed:
pythonPackages.ansible-lint: mark broken

old versions of ansible-lint are broken due to
setuptools no longer supporting their hacks.
Current versions of ansible-lint require newest
version of flit which isn't packaged yet

jonringer committed:
pythonPackages.altair: python2 no longer supported

jonringer committed:
Revert "python: ase: 3.17.0 -> 3.18.1"

This was a frozen version, wasn't meant to be bumped

jonringer committed:
pythonPackages.breathe: python2 no longer supported

jonringer committed:
pythonPackages.broadlink: add missing cryptography dep

jonringer committed:
pythonPackages.caldav: mark broken

requires xandikos package, which isn't in nixpkgs

jonringer committed:
pythonPackages.clize: fix tests

jonringer committed:
pythonPackages.colorcet: update tests

jonringer committed:
pythonPackages.fonttools: python2 no longer supported

veprbl committed:
pythonPackages.jedi: 0.15.1 -> 0.15.2

jonringer committed:
pythonPackages.azure-mgmt-servicefabric: fix build

jonringer committed:
python3Packages.azure-mgmt-dns: fix build

jonringer committed:
python3Packages.azure-mgmt-keyvault: fix build

jonringer committed:
devpi-server: fix tests

jonringer committed:
python3Packages.azure-mgmt-consumption: fix build

jonringer committed:
python3Packages.azure-mgmt-datamigration: fix build

jonringer committed:
python3Packages.azure-mgmt-devtestlabs: fix build

jonringer committed:
python3Packages.azure-mgmt-msi: fix build

jonringer committed:
python3Packages.azure-mgmt-relay: fix build

jonringer committed:
python3Packages.uproot: fix tests

jonringer committed:
pythonPackages.mozdevice: fix hash and build

jonringer committed:
pythonPackages.mozprofile: 2.3.0 -> 2.4.0

jonringer committed:
pythonPackages.keyring-alt: python2 no longer supported

jonringer committed:
pythonPackages.ppft: disable tests

tests are no longer packaged with pypi package

jonringer committed:
pythonPackages.partd: python2 no longer supported

jonringer committed:
pythonPackages.psd-tools: mark broken

missing packbits from nixpkgs

jonringer committed:
python35Packages.ipython: python35 not supported

jonringer committed:
python35Packages.ipython: freeze at 7.9.0

jonringer committed:
pythonPackages.django-contrib-comments: add six dep

jonringer committed:
pythonPackages.magic-wormhole-mailbox-server: disable python2 tests

unable to load zope.interface module

jonringer committed:
pythonPackages.magic-wormhole: disable python2 tests

unable to load zope.interface

jonringer committed:
pythonPackages.txtorcon: disable python2 tests

unable to load zope.interface

jonringer committed:
pythonPackages.monty: python2 no longer supported

jonringer committed:
pythonPackages.remotecv: mark broken

pillow no longer supports truncated images

jonringer commented:
disregard the interpreter bumps, already applied

jonringer commented:
starting to get to the point where the regressions are on leaf packages. I think this is ready to merge.

jonringer commented:
rebased on top of staging, there shouldn't be any conficts...

jonringer committed:
pythonPackages.nevow: fix build

jonringer commented:
most of the regressions are from python38Packages.aiohttp, https://hydra.nixos.org/jobset/nixpkgs/python-unstable

lilyball commented:
What's the status of this? `git-revise` is currently broken because of `importlib-metadata`.

jonringer commented:
Sorry, i've been visiting family and tied of with other things for the holidays. I'm assuming @FRidh has been as well

FRidh commented:
Yes indeed. I just got back so I can have a look at it again as well.

FRidh committed:
python: annoy: 1.16.2 -> 1.16.3

FRidh committed:
python: astunparse: 1.6.2 -> 1.6.3

FRidh committed:
python: awkward1: 0.1.32 -> 0.1.38

FRidh committed:
python: aws-sam-translator: 1.19.0 -> 1.19.1

FRidh committed:
python: aws-adfs: 1.21.1 -> 1.21.2

FRidh committed:
python: distributed: 2.9.0 -> 2.9.1

FRidh committed:
python: django-pglocks: 1.0.3 -> 1.0.4

FRidh committed:
python: beautifulsoup4: 4.8.1 -> 4.8.2

FRidh committed:
python: cfn-lint: 0.26.1 -> 0.26.2

FRidh committed:
python: browser-cookie3: 0.9.0 -> 0.9.1

FRidh committed:
python: azure-common: 1.1.23 -> 1.1.24

FRidh committed:
python: django-treebeard: 4.3 -> 4.3.1

FRidh committed:
python: CacheControl: 0.12.5 -> 0.12.6

FRidh committed:
python: dask: 2.9.0 -> 2.9.1

FRidh committed:
python: multidict: 4.7.1 -> 4.7.2

FRidh committed:
python: gym: 0.15.3 -> 0.15.4

FRidh committed:
python: mechanize: 0.4.4 -> 0.4.5

FRidh committed:
python: jupytext: 1.3.0 -> 1.3.1

FRidh committed:
python: dnslib: 0.9.10 -> 0.9.12

FRidh committed:
python: foolscap: 0.13.1 -> 0.13.2

FRidh committed:
python: holidays: 0.9.11 -> 0.9.12

FRidh committed:
python: mecab-python3: 0.996.2 -> 0.996.3

FRidh committed:
python: maxminddb: 1.5.1 -> 1.5.2

FRidh committed:
python: isbnlib: 3.9.9 -> 3.9.10

FRidh committed:
python: identify: 1.4.8 -> 1.4.9

FRidh committed:
python: keyring: 20.0.0 -> 20.0.1

FRidh committed:
python: eyeD3: 0.8.11 -> 0.8.12

FRidh committed:
python: ncclient: 0.6.6 -> 0.6.7

FRidh committed:
python: pybullet: 2.6.0 -> 2.6.1

FRidh committed:
python: py: 1.8.0 -> 1.8.1

FRidh committed:
python: pyfakefs: 3.7 -> 3.7.1

FRidh committed:
python: psd-tools: 1.8.34 -> 1.8.35

FRidh committed:
python: pymavlink: 2.4.2 -> 2.4.3

FRidh committed:
python: PyMuPDF: 1.16.9 -> 1.16.10

FRidh committed:
python: pymatgen: 2019.12.3 -> 2019.12.22

FRidh committed:
python: pyparsing: 2.4.5 -> 2.4.6

FRidh committed:
python: numpydoc: 0.9.1 -> 0.9.2

FRidh committed:
python: phonopy: 2.4.1.post5 -> 2.4.2

FRidh committed:
python: pyodbc: 4.0.27 -> 4.0.28

FRidh committed:
python: tox: 3.14.2 -> 3.14.3

FRidh committed:
python: runway-python: 0.5.4 -> 0.5.7

FRidh committed:
python: typeguard: 2.7.0 -> 2.7.1

FRidh committed:
python: qimage2ndarray: 1.8 -> 1.8.2

FRidh committed:
python: semantic_version: 2.8.3 -> 2.8.4

FRidh committed:
python: responses: 0.10.8 -> 0.10.9

FRidh committed:
python: pyvcd: 0.1.5 -> 0.1.6

FRidh committed:
python: pytest: 4.6.7 -> 4.6.8

FRidh committed:
python: slither-analyzer: 0.6.8 -> 0.6.9

FRidh committed:
python: SPARQLWrapper: 1.8.4 -> 1.8.5

FRidh committed:
python: shodan: 1.21.0 -> 1.21.1

FRidh committed:
python: regex: 2019.12.19 -> 2019.12.20

FRidh committed:
python: xml2rfc: 2.37.2 -> 2.37.3

FRidh committed:
python: uritemplate: 3.0.0 -> 3.0.1

FRidh committed:
python: zha-quirks: 0.0.28 -> 0.0.30

FRidh committed:
python: zeroconf: 0.24.2 -> 0.24.3

FRidh committed:
python: uncompyle6: 3.6.0 -> 3.6.1

FRidh commented:
Just did another rebase onto staging.

FRidh committed:
pythonPackages.sphinx: remove sqlalchemy

Not a mandatory dependency so remove it.

jtojnar committed:
python.pkgs.wrapPython: get rid of warning

When `makeWrapperArgs` variable is not set, `declare -p makeWrapperArgs`
will return with 1 and print an error message to stderr.

I did not handle the non-existence case in b0633406cb70e0e4ae3470a6b49e32b38d99ac16
because I thought `mk-python-derivation` will always define `makeWrapperArgs`
but `wrapProgram` can be called independently. And even with `mk-python-derivation`,
`makeWrappers` will not be set unless explicitly declared in the derivation
because of https://github.com/NixOS/nix/issues/1461.

I was lead to believe that because the builds were succeeding and I confirmed
that the mechanism fails when the variable is not defined and `-o nounset` is enabled.
It appears that `wrapPython` setup hook is not running under `-o nounset`, though,
invaldating the assumption.

Now we are checking that the variable exists before checking its type, which
will get rid of the warning and also prevent future error when `-o nounset`
is enabled in the setup hook.

For more information, see the discussion at
https://github.com/NixOS/nixpkgs/commit/a6bb2ede232940a96150da7207a3ecd15eb6328

jonringer commented:
I would like to see a lot of these changes in master. Not sure what to do to expedite this getting in

FRidh commented:
Tomorrow I intend to merge staging next at which point this goes into the
new staging next. If there aren't many other big changes that should land
in master relatively quickly.

On Sun, 29 Dec 2019, 21:44 Jon, <notifications@github.com> wrote:

> I would like to see a lot of these changes in master. Not sure what to do
> to expedite this getting in
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/76059?email_source=notifications&email_token=AAQHZ352XDH57MPG2MR7ON3Q3ED4JA5CNFSM4J5OFUN2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEHZH65Y#issuecomment-569540471>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQHZ33DZZU2MNYMCDPA2ELQ3ED4JANCNFSM4J5OFUNQ>
> .
>

jonringer committed:
nbstripout: fix build

FRidh committed:
pythonPackages.Babel: remove glibcLocales and don't set LC_ALL

as we already set

    LANG = "${if python.stdenv.isDarwin then "en_US" else "C"}.UTF-8";

in `buildPythonPackage`.

This is related to issue https://github.com/NixOS/nixpkgs/issues/74904

FRidh commented:
Forgot to merge it into staging so that's why it went to staging-next...

arnoldfarkas commented on pkgs/development/python-modules/gunicorn/default.nix:
```diff
@@ -8,11 +8,12 @@
 buildPythonPackage rec {
   pname = "gunicorn";
-  version = "19.9.0";
+  version = "20.0.4";
```
FYI this change breaks apache-airflow build:
```
ERROR: Could not find a version that satisfies the requirement gunicorn<20.0,>=19.5.0 (from apache-airflow==1.10.5) (from versions: none)
ERROR: No matching distribution found for gunicorn<20.0,>=19.5.0 (from apache-airflow==1.10.5)
```

jonringer replied:
opening a ticket is more action-able :), I unfortunately don't have time to look into this :(
